crates/ruff_graph/src/collector.rs

```diff
@@ -30,71 +30,99 @@ impl<'a> Collector<'a> {
-                    if let Some(module) = module {
-                        components.extend(module.split('.'));
-                    }
+    /// Helper to collect imports from an `import from` statement.
+    fn collect_import_from(&mut self, stmt: &ast::StmtImportFrom, type_checking: bool) {
+        let module = stmt.module.as_deref();
+        let level = stmt.level;
+        for alias in &stmt.names {
+            let mut components = vec![];
 
-                    // Add the alias name, unless it's a wildcard import.
-                    if alias.name.as_str() != "*" {
-                        components.push(alias.name.as_str());
-                    }
+            if level > 0 {
+                // If we're resolving a relative import, we must have a module path.
+                let Some(module_path) = self.module_path else {
+                    return;
+                };
 
-                    if let Some(module_name) = ModuleName::from_components(components) {
-                        self.imports.push(CollectedImport::ImportFrom(module_name));
+                // Start with the containing module.
+                components.extend(module_path.iter().map(String::as_str));
+
+                // Remove segments based on the number of dots.
+                for _ in 0..level {
+                    if components.is_empty() {
+                        return;
                     }
+                    components.pop();
                 }
             }
-            Stmt::Import(ast::StmtImport {
-                names,
-                range: _,
-                node_index: _,
-            }) => {
-                for alias in names {
-                    if let Some(module_name) = ModuleName::new(alias.name.as_str()) {
-                        self.imports.push(CollectedImport::Import(module_name));
+
+            // Add the module path.
+            if let Some(module) = module {
+                components.extend(module.split('.'));
+            }
+
+            // Add the alias name, unless it's a wildcard import.
+            if alias.name.as_str() != "*" {
+                components.push(alias.name.as_str());
+            }
+
+            if let Some(module_name) = ModuleName::from_components(components) {
+                self.imports.push(CollectedImport {
+                    module_name,
+                    import_kind: ImportKind::ImportFrom,
+                    type_checking,
+                });
+            }
+        }
+    }
+}
+
+impl<'ast> SourceOrderVisitor<'ast> for Collector<'_> {
+    fn visit_stmt(&mut self, stmt: &'ast Stmt) {
+        match stmt {
+            Stmt::ImportFrom(stmt) => {
+                self.collect_import_from(stmt, false);
+            }
+            Stmt::Import(stmt) => {
+                self.collect_import(stmt, false);
+            }
+            Stmt::If(ast::StmtIf { test, body, .. }) => {
+                if is_type_checking_condition(test) {
```

Comment:
I think a simpler approach is to pass the `exclude_type_checking` option to `Collector` and simply skip the entire body if `test` is a `TYPE_CHECKING` condition. 

This removes the need to split `collect_import_from` and `collect_import` out of `visit_stmt` or that we need to set `type_checking` on `CollectedImport`

Reply:
Oh yeah, that would be simpler. 

Then we don't track which imports are type-checking only for other uses. e.g., if we want to show only type-checking imports? 
Is that okay?